jlm/rvsdg/gamma.hpp

```diff
@@ -117,180 +116,184 @@ class GammaNode : public StructuralNode
 private:
   GammaNode(rvsdg::output * predicate, size_t nalternatives);
 
-  class entryvar_iterator
-  {
-  public:
-    constexpr entryvar_iterator(GammaInput * input) noexcept
-        : input_(input)
-    {}
-
-    GammaInput *
-    input() const noexcept
-    {
-      return input_;
-    }
-
-    const entryvar_iterator &
-    operator++() noexcept;
-
-    inline const entryvar_iterator
-    operator++(int) noexcept
-    {
-      entryvar_iterator it(*this);
-      ++(*this);
-      return it;
-    }
-
-    inline bool
-    operator==(const entryvar_iterator & other) const noexcept
-    {
-      return input_ == other.input_;
-    }
-
-    inline bool
-    operator!=(const entryvar_iterator & other) const noexcept
-    {
-      return !(*this == other);
-    }
-
-    GammaInput &
-    operator*() noexcept
-    {
-      return *input_;
-    }
-
-    GammaInput *
-    operator->() noexcept
-    {
-      return input_;
-    }
-
-  private:
-    GammaInput * input_;
+public:
+  /**
+   * \brief A variable routed into all gamma regions.
+   */
+  struct EntryVar
+  {
+    /**
+     * \brief Variable at entry point (input to gamma node).
+     */
+    rvsdg::input * input;
+    /**
+     * \brief Variable inside each of the branch regions (argument in per subregion).
+     */
+    std::vector<rvsdg::output *> branches;
```

Comment:
I really do not like the name `branches`. I find it very misleading. What about `subregionArgument` or even `branchArgument`?

Reply:
Or any other ideas? If everything fails, we can use `branches`, but I am not in favor or it.

Reply:
Some names that I came up with

- branchVar (this is the vesion of the variable in the respective branches)
- branchVariable, same as branchVar just longer
- branchDef (based on eventually renaming output->Def and input->Use)
- branchArg, shorter than branchArgument
- perBranch: actually quite accurately describes what it is, but at the same time sounds awkward

initially I had some dislike for the term "argument", but actually on thinking more about it "branchArg(ument)" (and "branchRes(ult)" for the converse sounds suitable -- generally I prefer not to abbreviate and would use the full names, but not certain. WDYT?

Reply:
changed it branchArgument & branchResult, I think that is appropriate

Reply:
branchArgument and branchResult sounds great.